src/panels/config/helpers/forms/ha-timer-form.ts

```diff
@@ -79,14 +82,14 @@ class HaTimerForm extends LitElement {
             "ui.dialogs.helper_settings.generic.icon"
           )}
         ></ha-icon-picker>
-        <ha-textfield
-          .configValue=${"duration"}
-          .value=${this._duration}
-          @input=${this._valueChanged}
-          .label=${this.hass.localize(
-            "ui.dialogs.helper_settings.timer.duration"
-          )}
-        ></ha-textfield>
+        <ha-time-input
+           .configValue=${"duration"}
+           .value=${this._duration}
+           .locale=${this.hass.locale}
```

Comment:
This will now show am/pm controls if the locale has those, which doesnt make sense for a duration.